packages/redux/src/raffles/__tests__/selectors.test.ts

```diff
@@ -0,0 +1,303 @@
+          participationId,
+        ),
+      ).toEqual(expectedRaffles);
+      expect(spy).toHaveBeenCalledWith(
+        mockRafflesState,
+        'raffleParticipations',
+        participationId,
+      );
+    });
+  });
+
+  describe('getRaffleParticipationForRaffle', () => {
+    it('should return a raffle participation', () => {
+      const expectedRaffleParticipation =
+        mockFetchRaffleParticipationsNormalizedPayload.entities
+          .raffleParticipations[participationId];
+      const spy = jest.spyOn(fromEntities, 'getEntities');
+      expect(
+        selectors.getRaffleParticipationForRaffle(
+          mockRafflesState as StoreState,
+          raffleId,
+        ),
+      ).toEqual(expectedRaffleParticipation);
+      expect(spy).toHaveBeenCalledWith(
+        mockRafflesState,
+        'raffleParticipations',
+      );
+    });
+
+    it('should return undefined when raffle participations are undefined', () => {
+      const spy = jest.spyOn(fromEntities, 'getEntities');
+      const state = { entities: {} } as StoreState;
+      expect(
+        selectors.getRaffleParticipationForRaffle(state, raffleId),
+      ).toBeUndefined();
+      expect(spy).toHaveBeenCalledWith(state, 'raffleParticipations');
+    });
+  });
+
+  describe('getDetailedRaffle', () => {
+    it('should return a detailed raffle', () => {
+      const expectedRaffles = {
+        ...mockFetchRaffleNormalizedPayload.entities.raffles[raffleId],
+        participation: mockRaffleParticipationResponse,
+        estimation: mockRaffleEstimationResponse,
+      };
+
+      const spy = jest.spyOn(fromEntities, 'getEntityById');
+      expect(
+        selectors.getDetailedRaffle(mockRafflesState as StoreState, raffleId),
+      ).toEqual(expectedRaffles);
+      expect(spy).toHaveBeenCalledWith(mockRafflesState, 'raffles', raffleId);
+      expect(spy).toHaveBeenCalledWith(
+        mockRafflesState,
+        'raffleEstimations',
+        raffleId,
+      );
+    });
+
+    it('should return undefined when raffle doesnt exist', () => {
+      const invalidRaffleId = 'invalid Raffle Id';
+      expect(
+        selectors.getDetailedRaffle(
+          mockRafflesState as StoreState,
+          invalidRaffleId,
+        ),
+      ).toBeUndefined();
+    });
+
+    it('should return just a raffle when raffle participation and raffle estimation do not exist', () => {
+      const expectedResult = {
+        ...mockFetchRaffleNormalizedPayload.entities.raffles[raffleId],
+        estimation: undefined,
+        participation: undefined,
+      };
+      const state = {
+        ...mockRafflesState,
+        entities: {
+          ...mockRafflesState.entities,
+          raffleParticipations: {},
+          raffleEstimations: {},
+        },
+      };
+      expect(
+        selectors.getDetailedRaffle(state as StoreState, raffleId),
+      ).toEqual(expectedResult);
+    });
+  });
+
+  describe('Loading and error state for All Raffles', () => {
```

Comment:
Use the same naming strategy you have used for the remaining selectors as these tests are no different than them.
For example this test should be declared as:

```js
describe('areAllRafflesLoading', () => {
   it('should return the loading status for the raffles request', () => {
     // test goes here...
   });
});

describe('getAllRafflesError', () => {
   it('should return the error status for the raffles request', () => {
     // test goes here...
   });
});
```

Please, do this for the remaining tests that are using the same strategy here.

Reply:
done, pls double check